pkgs/tools/security/pbis/default.nix

```diff
@@ -0,0 +1,69 @@
+{ stdenv, fetchFromGitHub, autoconf, automake, libtool, perl, flex, bison, curl,
+  pam, popt, libiconv, libuuid, openssl_1_0_2, cyrus_sasl, sqlite, tdb, libxml2 }:
+
+stdenv.mkDerivation rec {
+  pname = "pbis-open";
+  version = "9.1.0";
+
+  src = fetchFromGitHub {
+    owner = "BeyondTrust";
+    repo = pname;
+    rev = version;
+    sha256 = "081jm34sf488nwz5wzs55d6rxx3sv566x6p4h1yqcjaw36174m8v";
+  };
+
+  nativeBuildInputs = [
+    autoconf automake libtool perl flex bison
+  ];
+
+  # curl must be placed after openssl_1_0_2, because it pulls openssl 1.1 dependency.
+  buildInputs = [
+    pam popt libiconv libuuid openssl_1_0_2 cyrus_sasl
+    curl sqlite popt tdb libxml2 /*libglade2 for gtk*/
+  ];
+
+  postPatch = ''
+    patchShebangs .
+    sed -i -e 's/legacy//g' lwupgrade/MakeKitBuild # disable /opt/ symlinks
+    sed -i -e 's/tdb.h//g' samba-interop/MakeKitBuild #include <tdb.h> fails but it won't affect the build
+  '';
+  preConfigure = ''
+    mkdir release
+    cd release
+    NIX_CFLAGS_COMPILE="$NIX_CFLAGS_COMPILE -isystem ${libxml2.dev}/include/libxml2 -Wno-error=format-overflow -Wno-error=array-bounds -Wno-error=pointer-sign -Wno-error=deprecated-declarations -Wno-error=unused-variable"
+    configureFlags="
+      --docdir=$prefix/share/doc
+      --mandir=$prefix/share/doc/man
+      --datadir=$prefix/share
+      --lw-initdir=$prefix/etc/init.d
+      $configureFlags"
+  '';
+  configureScript = ''../configure'';
+  configureFlags = [
+    "CFLAGS=-O"
+    "--selinux=no" # NixOS does not support SELinux
+    "--build-isas=x86_64" # [lwbase] endianness (host/x86_32): [lwbase] ERROR: could not determine endianness
+    "--fail-on-warn=no"
+    # "--debug=yes"
+  ]; # ^ See https://github.com/BeyondTrust/pbis-open/issues/124
+  configureFlagsArray = [ "--lw-bundled-libs=linenoise-mob tomlc99 opensoap krb5 cyrus-sasl curl openldap" ];
+  # ^ it depends on old krb5 version 1.9 (issue #228)
+  # linenoise-mod, tomlc99, opensoap is not in nixpkgs.
+  # krb5 must be old one, and cyrus-sasl and openldap have dependency to newer libkrb5 that cause runtime error
+  enableParallelBuilding = true;
+  makeFlags = "SHELL=";
+  hardeningDisable = [ "format" ]; # -Werror=format-security
+  installPhase = ''
+    mkdir $sys
+    mv stage/{lib,var} $sys
+    mv stage$out $out
+  '';
+  outputs = [ "out" "sys" ];
+
+  meta = with stdenv.lib; {
+    description = "BeyondTrust AD Bridge Open simplifies the process of joining non-Microsoft hosts to Active Directory domains";
+    homepage = "https://github.com/BeyondTrust/pbis-open";
+    license = with licenses; [ gpl2 lgpl21 ];
+    platforms = with platforms; darwin ++ linux ++ freebsd;
```

Comment:
may want to make this this x86_64 variants of each:
```
configure flags: --prefix=/nix/store/57iw70y1vm0mznlkwzab32nz2564n0gy-pbis-open-9.1.0 --bindir=/nix/store/57iw70y1vm0mznlkwzab32nz2564n0gy-pbis-open-9.1.0/bin --sbindir=/nix/store/57iw70y1vm0mznlkwzab32nz2564n0gy-pbis-open-9.1.0/sbin --includedir=/nix/store/57iw70y1vm0mznlkwzab32nz2564n0gy-pbis-open-9.1.0/include --oldincludedir=/nix/store/57iw70y1vm0mznlkwzab32nz2564n0gy-pbis-open-9.1.0/include --mandir=/nix/store/57iw70y1vm0mznlkwzab32nz2564n0gy-pbis-open-9.1.0/share/man --infodir=/nix/store/57iw70y1vm0mznlkwzab32nz2564n0gy-pbis-open-9.1.0/share/info --docdir=/nix/store/57iw70y1vm0mznlkwzab32nz2564n0gy-pbis-open-9.1.0/share/doc/ --libdir=/nix/store/57iw70y1vm0mznlkwzab32nz2564n0gy-pbis-open-9.1.0/lib --libexecdir=/nix/store/57iw70y1vm0mznlkwzab32nz2564n0gy-pbis-open-9.1.0/libexec --localedir=/nix/store/57iw70y1vm0mznlkwzab32nz2564n0gy-pbis-open-9.1.0/share/locale --docdir=/nix/store/57iw70y1vm0mznlkwzab32nz2564n0gy-pbis-open-9.1.0/share/doc --mandir=/nix/store/57iw70y1vm0mznlkwzab32nz2564n0gy-pbis-open-9.1.0/share/doc/man --datadir=/nix/store/57iw70y1vm0mznlkwzab32nz2564n0gy-pbis-open-9.1.0/share --lw-initdir=/nix/store/57iw70y1vm0mznlkwzab32nz2564n0gy-pbis-open-9.1.0/etc/init.d CFLAGS=-O --selinux=no --build-isas=x86_64 --fail-on-warn=no --lw-bundled-libs=linenoise-mob\ tomlc99\ opensoap\ krb5\ cyrus-sasl\ curl\ openldap
           [makekit] initializing
          [platform] ERROR: unknown architecture: aarch64
builder for '/nix/store/b08ixkgkjw72liv64903msr9lgdccvbd-pbis-open-9.1.0.drv' failed with exit code 1
error: build of '/nix/store/b08ixkgkjw72liv64903msr9lgdccvbd-pbis-open-9.1.0.drv' failed
```

Reply:
nice catch!